astronomer/providers/apache/livy/hooks/livy.py

```diff
@@ -0,0 +1,410 @@
+        :return: response body
+        :rtype: dict
+        """
+        self.log.info("Fetching the logs for batch session with id: %d", session_id)
+        log_start_line = 0
+        log_total_lines = 0
+        log_batch_size = 100
+
+        while log_start_line <= log_total_lines:
+            # Livy log  endpoint is paginated.
+            result = await self.get_batch_logs(session_id, log_start_line, log_batch_size)
+            if result["status"] == "success":
+                log_total_lines = self._parse_request_response(result["response"], "total")
+                log_start_line += log_batch_size
+                log_lines = self._parse_request_response(result["response"], "log")
+                for log_line in log_lines:
+                    self.log.info(log_line)
+                return log_lines
+            else:
+                self.log.info(result["response"])
+                return result["response"]
+
+    @staticmethod
+    def _validate_session_id(session_id: Union[int, str]) -> None:
+        """
+        Validate session id is a int
+
+        :param session_id: session id
+        """
+        try:
+            int(session_id)
+        except (TypeError, ValueError):
+            raise TypeError("'session_id' must be an integer")
+
+    @staticmethod
+    def _parse_post_response(response: Dict[Any, Any]) -> Any:
+        """
+        Parse batch response for batch id
+
+        :param response: response body
+        :return: session id
+        :rtype: int
+        """
+        return response.get("id")
+
+    @staticmethod
+    def _parse_request_response(response: Dict[Any, Any], parameter: Any) -> Any:
+        """
+        Parse batch response for batch id
+
+        :param response: response body
+        :return: value of parameter
+        :rtype: Union[int, list]
+        """
+        return response.get(parameter)
+
+    @staticmethod
+    def build_post_batch_body(
+        file: str,
+        args: Optional[Sequence[Union[str, int, float]]] = None,
+        class_name: Optional[str] = None,
+        jars: Optional[List[str]] = None,
+        py_files: Optional[List[str]] = None,
+        files: Optional[List[str]] = None,
+        archives: Optional[List[str]] = None,
+        name: Optional[str] = None,
+        driver_memory: Optional[str] = None,
+        driver_cores: Optional[Union[int, str]] = None,
+        executor_memory: Optional[str] = None,
+        executor_cores: Optional[int] = None,
+        num_executors: Optional[Union[int, str]] = None,
+        queue: Optional[str] = None,
+        proxy_user: Optional[str] = None,
+        conf: Optional[Dict[Any, Any]] = None,
+    ) -> Any:
+        """
+        Build the post batch request body.
+        For more information about the format refer to
+        .. seealso:: https://livy.apache.org/docs/latest/rest-api.html
+        :param file: Path of the file containing the application to execute (required).
```

Comment:
I am sure this won't parse properly, can you post the screenshot of the rendered docs for this page

Reply:
Yes, it was not rendering correctly. [Changed](https://github.com/astronomer/astronomer-providers/pull/178/commits/a0452019dbad4dd5346389a3b06dcb3f99a023da) it